lib/src/main/java/oiccli/State.java

```diff
@@ -0,0 +1,194 @@
+package oiccli;
+
+import oiccli.exceptions.ExpiredToken;
+import oiccli.exceptions.UnknownState;
+import sun.swing.plaf.synth.DefaultSynthStyle;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class State {
+
+    private String clientId;
+    private Database db;
+    private int lifetime;
+
+    public State(String clientId, Database db, String dbName, int lifetime) {
+        this.clientId = clientId;
+        if (db == null) {
+            if (StringUtil.isNotNullAndNotEmpty(dbName)) {
+                this.db = shelve.open(dbName, true);
+            } else {
+                this.db = new Database();
+            }
+        } else {
+            this.db = db;
+        }
+        this.lifetime = lifetime;
+    }
+
+    public State(String clientId, Database db, String dbName) {
+        this(clientId, db, dbName, 600);
+    }
+
+    public String createState(final String receiver, Object request) {
+        String state = StringUtil.generateRandomString(24);
+        final long now = System.currentTimeMillis();
+        Map<String, Object> info = new HashMap<String, Object>() {{
+            put("clientId", clientId);
+            put("as", receiver);
+            put("iat", now);
+        }};
+        switch (state) {
```

Comment:
So `state` is a random 123abc string and you're comparing it against 1, 2 and 3?? this can't be right 🤔 . Also the switch is missing the "default" case.

Reply:
those were just placeholders.  i'm a bit confused how i'd implement it.  instances like these are where i'd appreciate your input.